src/main/scala/cognite/spark/v1/PushdownUtilities.scala

```diff
@@ -59,7 +59,11 @@ object PushdownUtilities {
   // scalastyle:off
   def getFilter(filter: Filter): PushdownExpression =
     filter match {
-      case IsNotNull(colName) => NoPushdown()
+      case IsNotNull(_) | IsNull(_) | EqualNullSafe(_, null) => NoPushdown()
+      case EqualTo(_, null) | GreaterThan(_, null) | GreaterThanOrEqual(_, null) | LessThan(_, null) |
+          LessThanOrEqual(_, null) =>
+        throw new Exception(
+          "Wait, this should not happen... This is always false and Spark should just optimize it away.")
```

Comment:
But just in case ;) "Unexpected error, please contact support@cognite.com and tell them Spark is misbehaving."
After all, a user *could* end up seeing this message!